rust/xaynet-analytics/src/controller.rs

```diff
@@ -121,18 +146,39 @@ impl AnalyticsController {
         )
     }
 
-    // TODO: review and debug this method during https://xainag.atlassian.net/browse/XN-1560
+    /// This method implements a sliding 'time window' of self.send_data_frequency duration, to check whether we have
+    /// already sent data in the current window, or not.
+    ///
+    /// An alternative implementation could be based on simply checking whether:
+    /// last_time_data_sent > Utc::now() - self.send_data_frequency
+    ///
+    /// In the current implementation, it might be easier to then group the aggregated data on the coordinator side,
+    /// to then be displayed in the UI, especially if self.send_data_frequency = Duration::hours(24).
+    ///
+    /// The more dynamic approach however implies that if, for example, self.send_data_frequency = Duration::hours(6),
+    /// and the last time we sent the data was at 5AM, we would be able to send again at 7AM, while with the simpler solution
+    /// we wouldn't be able to send again until 11AM.
+    ///
+    /// The correct approach to be chosen should very much depend on the amount of data available for aggregation,
+    /// and it's possible that Self::MAX_SEND_DATA_FREQUENCY_HOURS should be increased to more than 24.
```

Comment:
this sentence probably needs a few extra words because as it stands, I think the current implementation is rather coupled to `MAX_SEND_DATA_FREQUENCY_HOURS` being `24`: the first period always begins at the start (midnight) of _today_
```rust
                let now = Utc::now();
                let start_of_day: DateTime<Utc> = DateTime::from_utc(
                    NaiveDate::from_ymd(now.year(), now.month(), now.day()).and_hms(0, 0, 0),
                    Utc,
                );
```
and therefore the last period ends by midnight _tomorrow_! so we only have 24 hours to play with and not more than that. so we ought to mention in the docstring that for higher values, we'll probably need to tweak this method to allow it.

Reply:
yes! will do that. i think i mentioned it either in confluence or in a comment on github, but not here. i'll be more clear, thanks